spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -1207,7 +1261,29 @@ public void run() {
 		}
 	}
 
+	private class IdleContainerStopper implements Runnable {
+
+		@Override
+		public void run() {
+			synchronized(JmsOutboundGateway.this.lifeCycleMonitor) {
+				if (System.currentTimeMillis() - lastSend > idleReplyContainerTimeout
+						&& replies.size() == 0) {
+					if (logger.isDebugEnabled()) {
+						logger.debug(getComponentName() + ": Stopping idle reply container.");
+					}
+					if (replyContainer.isRunning()) {
```

Comment:
Looks like the previous `DEBUG` must be moved here: it does not make sense to say `Stopping idle reply container.` if we really don't stop it according to this `isRunning()` condition